subsys/bluetooth/host/adv.c

```diff
@@ -462,20 +492,30 @@ static int hci_set_adv_ext_fragmented(struct bt_le_ext_adv *adv, uint16_t hci_op
 
 			/* We can't necessarily set one AD field in a single step. */
 			while (offset < data[j].data_len) {
-				buf = bt_hci_cmd_create(hci_op, sizeof(*set_data));
+				BUILD_ASSERT(sizeof(struct bt_hci_cp_le_set_per_adv_data)
+					     == UINT8_MAX);
+				BUILD_ASSERT(sizeof(struct bt_hci_cp_le_set_ext_adv_data)
+					     == UINT8_MAX);
+				BUILD_ASSERT(sizeof(struct bt_hci_cp_le_set_ext_scan_rsp_data)
+					     == UINT8_MAX);
+				buf = bt_hci_cmd_create(hci_op, UINT8_MAX);
```

Comment:
Perhaps a `MAX` of the `sizeof` the 3 structs would be better to use than `UINT8_MAX`. Alternative create a lookup size by `hci_op`.

Reply:
If you're ok with it, I would just keep it as is. It is just a matter of style. The correctness is ensured by the build assertions

Reply:
I have a slight preference to using MAX and sizeof (which also adds some semantic reasoning behind the value), but with the BUILD_ASSERTs I'm ok with this.